src/test/java/pl/kl/weatherservice/location/LocationsServiceTest.java

```diff
@@ -0,0 +1,125 @@
+        // then
+        assertThat(exception).isInstanceOf(EmptyInputException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenCountryIsEmpty_throwsEmptyInputException() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", "", 54.3, 18.6));
+        // then
+        assertThat(exception).isInstanceOf(EmptyInputException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenCountryIsNull_throwsEmptyInputException() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", null, 54.3, 18.6));
+        // then
+        assertThat(exception).isInstanceOf(EmptyInputException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenLatitudeIsMoreThanLimit_throwsInputOutOfRangeException() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", "Poland", 150.0, 18.6));
+        // then
+        assertThat(exception).isInstanceOf(InputOutOfRangeException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenLatitudeIsLessThanLimit_createsNewLocation() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", "Poland", -150.0, 18.6));
+        // then
+        assertThat(exception).isInstanceOf(InputOutOfRangeException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenLongitudeIsMoreThanLimit_throwsInputOutOfRangeException() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 200.0));
+        // then
+        assertThat(exception).isInstanceOf(InputOutOfRangeException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenLongitudeIsLessThanLimit_throwsInputOutOfRangeException() {
+        // when
+        Throwable exception = catchThrowable(() -> locationsService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, -200.0));
+        // then
+        assertThat(exception).isInstanceOf(InputOutOfRangeException.class);
+        verify(locationsRepository, times(0)).save(any());
+    }
+
+    @Test
+    void createLocation_whenRegionIsEmpty_createsNewLocation() {
+        // given
+        when(locationsRepository.save(any())).thenReturn(LocationTestHelper.LOCATION_COCHABAMBA);
+        // when
+        Location location = locationsService.createLocation("Cochabamba", null, "Bolivia", -17.4, -66.1);
```

Comment:
let's add a test with `"   "` value for a region

Reply:
Fixed. Test updated.